### PR TITLE
feat: emoji support per font and cdn

### PR DESCRIPTION
**Pull request by Flo0806**
## Summary

Now we can use emojis per font and per CDN. Integrated COLR v0/v1 renders emojis directly. 

## Type of change

- [ ] Bug fix
- [x] Feature
- [ ] Refactor
- [ ] Docs
- [ ] CI / tooling

## Checklist

- [x] `pnpm exec vitest run` is green
- [ ] No breaking changes (or called out above)
- [x] Docs updated if needed

## Summary by CodeRabbit

* **New Features**
  * Added support for color emoji rendering from either embedded fonts or image sources.
  * Introduced gradient-filled vector paths, expanding rich text and graphics output.
  * Improved font parsing to handle more Unicode glyphs and color-font formats.

* **Bug Fixes**
  * Fixed emoji and truncation handling so astral characters are measured and trimmed safely.
  * Improved PDF output so gradient and emoji resources are included correctly.
  * Scoped release notes and contributor reporting to the relevant package only.

* **Tests**
  * Added coverage for color glyphs, emoji images, gradients, and Unicode edge cases.

### CODE REVIEW

File: src/lib/renderer/pdf-backend.ts [hunk 1]
```diff
@@ -218,6 +236,28 @@ export class PdfBackend {
...
+        if (!(node.fill instanceof Color)) {
+          // Gradient: clip to the path (W n keeps it as the clip without painting it), then paint
+          // the registered shading across that clip. The q/Q isolates the clip.
+          const shading = om.registerShading(node.fill);
+          return `q\n${path}W n\n/${shading} sh\nQ\n`;
+        }
+        const body = `${node.fill.toPDFColorString()} rg\n${path}f\n`;
+        const gs = PdfBackend.alphaPrefix(om, node.fill.getAlpha(), 1);
+        return gs ? `q\n${gs}${body}Q\n` : body;
+      }
```
[the remark this answers is not in the document]
Comment (Flo0806) [reply src/lib/renderer/pdf-backend.ts:274]: understand. but COLR gradient stops are opak in real fonts = renders correctly. But thx!

File: src/lib/renderer/text-renderer.ts [hunk 2]
```diff
@@ -93,6 +102,200 @@ export class TextRenderer {
...
+        // drawing a broken glyph.
+        flushPending();
+        const node = await emojiImageNode(cp, imageSource, cursorX, run.y, run.fontSize);
+        if (node) nodes.push(node);
+      } else {
+        if (!pending) pendingX = cursorX;
+        pending += ch;
+      }
+      cursorX += advance;
+    }
```
[the remark this answers is not in the document]
Comment (Flo0806) [reply src/lib/renderer/text-renderer.ts:209]: Yes - you're right! But this is by design..
